File: src/types/medicalRecord.ts
export interface Patient {
  patientId: string
  id?: number | string
  patientCode?: string
  patientIdCode?: string
  fullName: string
  email?: string
  phone?: string
  phoneNumber?: string
  dateOfBirth?: string
  gender?: string
  address?: string
  citizenId?: string
  bloodType?: string
  allergyNote?: string | null
  allergies?: string
  medicalHistory?: string | null
  status?: string
  createdAt?: string
  updatedAt?: string
}

export interface MedicalRecord {
  id?: number
  medicalRecordId?: number
  recordId?: string
  medicalRecordCode?: string
  medicalRecordIdCode?: string
  recordIdCode?: string
  visitId?: number
  patientId: string
  patientCode?: string
  patientIdCode?: string
  patientName?: string
  appointmentId?: string
  doctorId?: string | number
  doctorName?: string
  chiefComplaint?: string
  symptoms?: string
  vitalSignsJson?: string
  diagnosis?: string
  diagnosisCode?: string
  diagnosisText?: string
  doctorNotes?: string
  doctorNote?: string
  treatmentPlan?: string
  followUpDate?: string
  status?: string
  examDate?: string
  createdAt?: string
  updatedAt?: string
  completedAt?: string
}

File: backend/MedicalAPI/Domain/Entities/MedicalRecord.cs
using MedicalAPI.Domain.Constants;

namespace MedicalAPI.Domain.Entities;

public sealed class MedicalRecord
{
    public int Id { get; set; }
    public string? MedicalRecordCode { get; set; }
    public int VisitId { get; set; }
    public int PatientId { get; set; }
    public int DoctorId { get; set; }
    public string? DiagnosisCode { get; set; }
    public string? DiagnosisSpecialty { get; set; }
    public string DiagnosisText { get; set; } = string.Empty;
    public string? DoctorNote { get; set; }
    public string? TreatmentPlan { get; set; }
    public DateOnly? FollowUpDate { get; set; }
    public string Status { get; set; } = MedicalStatuses.Draft;
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
    public DateTime? UpdatedAt { get; set; }
    public DateTime? CompletedAt { get; set; }
}
